homeassistant/components/todoist/calendar.py

```diff
@@ -296,6 +296,7 @@ def __init__(
         )
         self._cal_data = {}
         self._name = data[CONF_NAME]
+        self._attr_unique_id = data[ID]
```

Comment:
This is the ID of the calendar as returned by the API and is stable.

Reply:
I don't see this key in the data passed to the entity.

Please also add a test.

Reply:
It's not obvious from this diff, but it happens here: https://github.com/home-assistant/core/blob/fcbe2cc44be0172153a272fa43e72eb18bb3f790/homeassistant/components/todoist/calendar.py#L140-L141

I switched to the constant that is being used above, even though they are the same values for consistency.

Reply:
And how does it work for the extra projects?

https://github.com/home-assistant/core/blob/0df08b6b0ce6e8d853ba4981e8f1e1492227a198/homeassistant/components/todoist/calendar.py#L169-L180

Reply:
That's a great point!  Thanks for noticing that. Since the user configures these via yaml, it's probably safe to return None for those entities `unique_id`. I updated setting the unique id attribute to take that into account.